lib/LANraragi/Plugin/Metadata/CopyArchiveTags.pm

```diff
@@ -0,0 +1,88 @@
+package LANraragi::Plugin::Metadata::CopyArchiveTags;
+
+use strict;
+use warnings;
+
+use LANraragi::Model::Plugins;
+use LANraragi::Utils::Database qw(get_archive_tags);
+use LANraragi::Utils::Logging  qw(get_plugin_logger);
+use LANraragi::Utils::Tags     qw(join_tags_to_string);
+
+#Meta-information about your plugin.
+sub plugin_info {
+
+    return (
+        #Standard metadata
+        name        => "CopyArchiveTags",
+        type        => "metadata",
+        namespace   => "copy-archive-tags",
+        author      => "IceBreeze",
+        version     => "1.0",
+        description => "Copy tags from another LRR archive given either the URI or the ID.",
+        parameters  => [
+            {   type => "bool",
+                name => 'copy_date_added',
+                desc => "Enable to also copy the date (but it's up to you to remove the old one)"
+            }
+        ],
+        oneshot_arg => "LRR Gallery URI or ID:"
+    );
+
+}
+
+sub get_tags {
+    my $params = read_params(@_);
+    my $logger = get_plugin_logger();
+
+    # should be handled in the caller
+    my %hashdata = eval { internal_get_tags( $logger, $params ); };
+    if ($@) {
+        $logger->error($@);
+        return ( error => $@ );
+    }
+
+    $logger->info( "Sending the following tags to LRR: " . ( $hashdata{tags} || '-' ) );
+    return %hashdata;
+}
+
+sub internal_get_tags {
+    my ( $logger, $params ) = @_;
+
+    my $lrr_gid = $params->{'oneshot'};
```

Comment:
This doesn't seem to have any fail states in case an invalid ID is provided (or no ID at all, which could happen if you enable this plugin for automatic execution for some reason)

Reply:
It does nothing unless the ID matches something, but we can definitely avoid the call.